### PR TITLE
Tags Backup & Restore

### LINKED ISSUE #2863: [FEATURE] Tags
### Please confirm the following:

- [X] I've checked [the current issues](https://github.com/Ivy-Apps/ivy-wallet/issues) for duplicate issues.
- [X] I've requested a single (only one) feature/change in this issue. It complies with the One Request Per GitHub Issue (ORPGI) rule.
- [X] My issue is well-defined and describes how it should be implemented from UI/UX perspective.

### What do you want to be added or improved?

As a user I want to be able to tag certain transactions to better manage my finances.

### Why do you need it?

Helps to segregate finances better. 

### How do you imagine it?

Phase 1
1) CRUD on Tag Entities
2) Ability to add multiple tags to transactions.
3) Report Filter Based on tags

Phase 2
1) Optional Separate screen for tags to view transactions associated with it quickly
2) Search based on tags

### PR DESCRIPTION
**Pull request by Vishwa-Raghavendra**
## Pull Request (PR) Checklist
Please check if your pull request fulfills the following requirements:
- [x] The PR is submitted to the `main` branch.
- [x] I've read the [Contribution Guidelines](https://github.com/Ivy-Apps/ivy-wallet/blob/main/CONTRIBUTING.md) and my PR doesn't break the rules.
- [x] I've read the [Architecture Guidelines](https://github.com/Ivy-Apps/ivy-wallet/blob/main/docs/Architecture.md).
- [x] I confirm that I've run the code locally and everything works as expected.
- [x] 🎬 I've attached a **screen recoding** of the changes. 

> Tip: drag & drop the video to the PR description.

## What's changed?

Describe with a few bullets **what's new:**
- a Added Fix Which causes tag not to be selected in EditTransactionScreen
- b UX Fix for FilterOverly in ReportsScreen
- c BackUp and Restore for Tags

> 💡 Tip: Please, attach screenshots and screen recordings. It helps a lot!

## Risk Factors

**What may go wrong if we merge your PR?**

- a
- b

**In what cases your code won't work?**

- a
- b

## Does this PR closes any GitHub Issues?

Check **[Ivy Wallet Issues](https://github.com/Ivy-Apps/ivy-wallet/issues)**.

- Closes #2863 

> Replace `ISSUE_NUMBER` with the number of the GitHub issue that this PR is fixing. If you've done that correctly, you'll see the issue title linked when previewing your PR description.

> **!Note:** Do **not** link the PR number. Link the number/id of the GitHub Issue from [issues](https://github.com/Ivy-Apps/ivy-wallet/issues).

## Troubleshooting CI failures ❌

GitHub Actions failing? Read our [CI Troubleshooting guide](https://github.com/Ivy-Apps/ivy-wallet/blob/main/docs/CI-Troubleshooting.md).

### CODE REVIEW

**Review by ILIYANGERMANOV — APPROVED** [review]
Let's use the repository for the backup use-case

File: .idea/kotlinScripting.xml
Comment (ILIYANGERMANOV) [inline .idea/kotlinScripting.xml:1]: Add this file to `.gitignore` or revert changing it

Comment (Vishwa-Raghavendra) [reply .idea/kotlinScripting.xml:1]: Ah Sorry Missed this

**Review by ILIYANGERMANOV — APPROVED** [review]
Nice 💯

**Pull request merged**